bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/zk/ZKMetadataDriverBase.java

```diff
@@ -60,12 +60,16 @@
 
     protected static final String SCHEME = "zk";
 
-    public static String getZKServersFromServiceUri(URI uri) {
-        return uri.getAuthority().replace(";", ",");
+    public static String getZKServersFromServiceUri(URI uri) throws IllegalArgumentException {
+        String authority = uri.getAuthority();
+        if (authority == null) {
+            throw new IllegalArgumentException("Invalid metadata service URI format: " + uri);
+        }
+        return authority.replace(";", ",");
     }
 
     @SuppressWarnings("deprecation")
-    public static String resolveZkServers(AbstractConfiguration<?> conf) {
+    public static String resolveZkServers(AbstractConfiguration<?> conf) throws IllegalArgumentException {
```

Comment:
who is calling this method ? "resolveZkServers"
we should catch IllegalArgumentException as well

Reply:
I checked the usages of resolveZkServers and I think it is ok to keep IAE unwrapped